### ISSUE TITLE
Astro v2.4.5 breaks use of local icons when Astro assets enabled

### ISSUE THREAD

**Issue by paulrudy** [natemoo-re/astro-icon#95]
I'm using Astro in SSR mode with the Vercel adapter. After upgrading from Astro v2.4.4 to v2.4.5, I get the following error:

```
Error: [astro-icon] Unable to load "/src/icons/logo.svg". Does the file exist?
    at Module.load [as default] (/node_modules/.pnpm/astro-icon@0.8.0/node_modules/astro-icon/lib/utils.ts:186:13)
    at eval (/node_modules/.pnpm/astro-icon@0.8.0/node_modules/astro-icon/lib/Icon.astro:22:52)
    at AstroComponentInstance.Icon [as factory] (/node_modules/.pnpm/astro@2.4.5_sass@1.62.1_sharp@0.32.1/node_modules/astro/dist/runtime/server/astro-component.js:22:12)
    at AstroComponentInstance.init (/node_modules/.pnpm/astro@2.4.5_sass@1.62.1_sharp@0.32.1/node_modules/astro/dist/runtime/server/render/astro/instance.js:28:29)
    at AstroComponentInstance.render (/node_modules/.pnpm/astro@2.4.5_sass@1.62.1_sharp@0.32.1/node_modules/astro/dist/runtime/server/render/astro/instance.js:33:18)
    at render.next (<anonymous>)
    at Module.renderChild (/node_modules/.pnpm/astro@2.4.5_sass@1.62.1_sharp@0.32.1/node_modules/astro/dist/runtime/server/render/any.js:30:18)
    at async [Symbol.asyncIterator] (/node_modules/.pnpm/astro@2.4.5_sass@1.62.1_sharp@0.32.1/node_modules/astro/dist/runtime/server/render/astro/render-template.js:34:7)
    at async Module.renderAstroTemplateResult (/node_modules/.pnpm/astro@2.4.5_sass@1.62.1_sharp@0.32.1/node_modules/astro/dist/runtime/server/render/astro/render-template.js:42:20)
    at async Module.renderChild (/node_modules/.pnpm/astro@2.4.5_sass@1.62.1_sharp@0.32.1/node_modules/astro/dist/runtime/server/render/any.js:28:5)
```

Adding `"svgo": "2.8.0"` to devDependencies [as suggested here](https://github.com/natemoo-re/astro-icon/issues/93#issuecomment-1540191143) didn't help.

Both of the commits in Astro v2.4.5 relate to paths.

**Comment by paulrudy** [comment]
Updated the title of the issue: This issue only relates to local icons, and it seems to have to do with Astro assets (so perhaps [this commit](https://github.com/withastro/astro/commit/4f1073a6a4f3e5a4fc9df96a2ae59f2e929703fe) from v 2.4.5. When I disabled the experimental assets feature, this issue was resolved. It would be nice to be able to continue to use Astro assets, however.

**Comment by stramel** [comment]
I'll try to take a look tonight because I was just using it. I might be on 2.4.4 though

**Comment by stramel** [comment]
Can confirm after upgrading to 2.4.5, it is broken. Looking into a fix

**Comment by stramel** [comment]
Appears to be an upstream issue. Filed a upstream bug: https://github.com/withastro/astro/issues/7082

**Comment by paulrudy** [comment]
> Appears to be an upstream issue. Filed a upstream bug: https://github.com/withastro/astro/issues/7082

Sweet, thank you for looking into it!

**Comment by richardk80** [comment]
Are you using "experimental: { assets: true }" in the astro.config.mjs file by chance?

**Comment by paulrudy** [comment]
> Are you using "experimental: { assets: true }" in the astro.config.mjs file? If so...remove it.

This issue is meant to report specially that there is a conflict between the experimental assets and astro-icon since Astro v3.4.5. Removing the use of the experimental assets would certainly be a way not to encounter this issue. So would uninstalling astro-icon.

I'm interested in them both playing nicely together again.

**Comment by paulrudy** [comment]
Fixed with Astro v2.5.0!
